openhands/controller/agent_controller.py

```diff
@@ -72,6 +72,7 @@ def __init__(
         initial_state: State | None = None,
         is_delegate: bool = False,
         headless_mode: bool = True,
+        in_asyncio: bool = True,
```

Comment:
I introduced this flag to mark whether the existing behaviour should continue or not. In the existing behavior, a task is created on the current run loop by the constructor. In the new behavior, we do this later.

Reply:
I didn't know whether this would affect CLI Mode / Interactive Console / the Github Resolver - so I acted defensively.